src/Simulation/Simulators/OpenSystemsSimulator/DataModel/Generator.cs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using NumSharp;
+using static System.Math;
+
+namespace Microsoft.Quantum.Experimental;
+
+public record class GeneratorCoset(
```

Comment:
Does "Coset" mean something like a "co-set" (addendum to a set, complementary set)?

Reply:
A "coset" is the set obtained by multiplying all elements of another set by a given element. For example, if 𝑆 = {𝑠₀, 𝑠₁, …}, then the left coset 𝑎𝑆 = {𝑎𝑠₀, 𝑎𝑠₁, …}, the right coset 𝑆𝑏 = {𝑠₀𝑏, 𝑠₁𝑏, …}, and one can consider cosets in the most general form of 𝑎𝑆𝑏 = {𝑎𝑠₀𝑏, 𝑎𝑠₁𝑏, …}.

Here, a generator coset is used to denote that a generator 𝐺 = −𝑖𝐿 + 𝐷 generates a continuous time monoid Λ(𝑡) = 𝑒^{𝑡𝐺} such that noise of the form Φ_post Λ(𝑡) Φ_pre is a coset of the monoid given by the generator 𝐺. Colloquially, this captures noise where some fixed time-independent thing may or may not happen, a time dependent thimg happens, and then a second fixed time-independent thing may or may not happen.